### PR TITLE
Storage bug fixes

### PR DESCRIPTION
**Pull request by princiya**
Fixes the following bugs:

Bug: #48 
Bug: #50  
Bug #51 

In addition to the above bug fixes, third parties are also displayed.
screen shot 2017-06-16 at 14 48 30

### CODE REVIEW

File: js/store.js [hunk 1]
```diff
@@ -43,7 +51,15 @@ const store = {
     }
 
     const websites = clone(this._websites);
-    websites[hostname] = data;
+
+    if(!websites[hostname]) {
+      websites[hostname] = {};
+    }
+
+    for(const key in data) {
```
Comment (jonathanKingston) [inline js/store.js]: Nit: space before `(`, we should have an ESLint rule for this too.

Comment (princiya) [inline js/store.js]: oh yes, lot of inconsistencies here for `if` and `for` statements. sending a fix.

File: js/store.js [hunk 2]
```diff
@@ -2,8 +2,15 @@ const store = {
   _websites: null,
 
   async init() {
-    const data = await browser.storage.local.get('websites');
-    this._websites = clone(data.websites);
+    if(!this._websites || isObjectEmpty(this._websites)) {
+      const data = await browser.storage.local.get('websites');
+
+      if(!data.websites) {
```
Comment (jonathanKingston) [inline js/store.js]: Nit: space before `(`

File: js/store.js [hunk 3]
```diff
@@ -43,7 +51,15 @@ const store = {
     }
 
     const websites = clone(this._websites);
-    websites[hostname] = data;
+
+    if(!websites[hostname]) {
```
Comment (jonathanKingston) [inline js/store.js]: Nit: space before `(`

File: js/store.js (hunk 1 above)
Comment (princiya) [inline js/store.js]: done. but will create an eslint rule. creating an issue

File: js/viz.js [hunk 4]
```diff
@@ -4,13 +4,22 @@ const viz = {
...
     context.beginPath();
 
     for(const website in websites) {
       context.moveTo(x, y);
-      context.arc(x, y, 5, 0, pi);
+      context.arc(x, y, 10, 0, pi);
       context.fillText(website, x, y);
+
+      let x1 = x;
+      for(const thirdParty in websites[website].thirdPartyRequests) {
```
Comment (jonathanKingston) [inline js/viz.js]: :metal:

**Pull request merged**